src/backend/binaryen/utils.ts

```diff
@@ -1778,31 +1779,83 @@ export namespace FunctionalFuncs {
             case ts.SyntaxKind.EqualsEqualsEqualsToken:
             case ts.SyntaxKind.ExclamationEqualsToken:
             case ts.SyntaxKind.ExclamationEqualsEqualsToken: {
-                if (rightValueType.kind === ValueTypeKind.NULL) {
-                    res = module.call(
-                        dyntype.dyntype_is_null,
-                        [dynCtx, leftValueRef],
-                        binaryen.i32,
-                    );
+                if (
+                    leftValueType.kind === ValueTypeKind.NULL ||
+                    rightValueType.kind === ValueTypeKind.NULL
+                ) {
+                    if (leftValueType.kind === ValueTypeKind.NULL) {
+                        res = module.call(
+                            dyntype.dyntype_is_null,
+                            [dynCtx, rightValueRef],
+                            binaryen.i32,
+                        );
+                    } else {
+                        res = module.call(
+                            dyntype.dyntype_is_null,
+                            [dynCtx, leftValueRef],
+                            binaryen.i32,
+                        );
+                    }
                     // TODO: ref.null need table.get support in native API
-                } else if (rightValueType.kind === ValueTypeKind.UNDEFINED) {
-                    res = isDynUndefined(module, leftValueRef);
-                } else if (rightValueType.kind === ValueTypeKind.NUMBER) {
-                    res = operateF64F64ToDyn(
-                        module,
-                        leftValueRef,
-                        rightValueRef,
-                        opKind,
-                        true,
-                    );
-                } else if (rightValueType.kind === ValueTypeKind.STRING) {
-                    res = operateStrStrToDyn(
-                        module,
-                        leftValueRef,
-                        rightValueRef,
-                        opKind,
-                        true,
-                    );
+                } else if (
+                    leftValueType.kind === ValueTypeKind.UNDEFINED ||
+                    rightValueType.kind === ValueTypeKind.UNDEFINED
+                ) {
+                    if (leftValueType.kind === ValueTypeKind.UNDEFINED) {
+                        res = isDynUndefined(module, rightValueRef);
+                    } else {
+                        res = module.call(
+                            dyntype.dyntype_is_undefined,
+                            [dynCtx, leftValueRef],
+                            binaryen.i32,
+                        );
```

Comment:
Same as above, and `isDynUndefined` can be used directly.

Reply:
Done